### ISSUE TITLE
Testing the model on data used to train it?

### ISSUE THREAD

**Issue by nyck33** [aws-samples/aws-machine-learning-university-accelerated-tab#19]
_One thing we can do is to test the model with the data we used to train it, and use sklearn's metrics functions to examine the performance of the classifier._

Isn't this wrong?  Shouldn't the data set be divided into training and validation sets?

**Comment by uditgt** [comment]
Conceptually yes, validation set should be separated from training data. But many sklearn functions allow us to run cross-validation inherently as part of their fitting procedure - so we don't need to create a separate validation set; its created as part of CV process. We can feed the model the entire training data.
